mcp-test/src/main/java/io/modelcontextprotocol/client/AbstractMcpAsyncClientTests.java

```diff
@@ -92,8 +96,16 @@ void testConstructorWithInvalidArguments() {
 			.hasMessage("Request timeout must not be null");
 	}
 
+	@Test
+	void testListToolsWithoutInitialization() {
+		assertThatThrownBy(() -> mcpAsyncClient.listTools(null).block()).isInstanceOf(McpError.class)
```

Comment:
The test will last at least 600ms in case of `WebClient` and 4s for JDK `HttpClient` before the exception is observed, correct? If I am correct, it would be useful to consider using StepVerifier.withVirtualTime to emulate time passing by instead of adding more seconds to the time w validate the behaviour.